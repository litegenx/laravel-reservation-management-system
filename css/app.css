.router-link-active {
  cursor: pointer;
}

