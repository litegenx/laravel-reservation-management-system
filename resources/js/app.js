require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
