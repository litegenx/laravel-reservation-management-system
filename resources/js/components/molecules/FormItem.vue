<template>
    <div>
        <component
            :is="componentIs"
            :form="form"
            :detail="detail"
            :form-inputs="formInputs"
            :value="value"
            :label="label"
            :hint="hint"
            :validate-errors="validateErrors"
            @input="val => $emit('input', val)"
        />
    </div>
</template>

<script>
    import { get } from 'lodash';
    import TextForm from './form/Text';
    import DateForm from './form/Date';
    import SearchForm from './form/Search';

    export default {
        components: {
            TextForm,
            DateForm,
            SearchForm,
        },
        props: {
            detail: {
                type: Object,
                required: true,
            },
            formInputs: {
                type: Object,
                required: true,
            },
            form: {
                type: Object,
                required: true,
            },
            value: {
                type: [String, Number],
                required: true,
            },
            validateErrors: {
                type: Array,
                required: true,
            },
            increment: {
                type: Number,
                required: true,
            },
        },
        computed: {
            name () {
                return this.form.text;
            },
            label () {
                return this.$t('column.' + this.name);
            },
            type () {
                return this.form.type || 'text';
            },
            hint () {
                return this.form.hint ? this.$t(this.form.hint) : '';
            },
            search () {
                return this.form.search || '';
            },
            componentIs () {
                if ('date' === this.type) {
                    return DateForm;
                }
                if ('search' === this.type) {
                    return SearchForm;
                }
                return TextForm;
            },
        },
    };
</script>
