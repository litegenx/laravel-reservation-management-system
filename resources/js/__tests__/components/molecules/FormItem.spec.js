import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import FormItem from '../../../components/molecules/FormItem';
import app from '../app';
import setupLocalVue from '../setupLocalVue';

jest.useFakeTimers();

describe('FormItem', () => {
    it('should not render date form', () => {
        const wrapper = mount(app(FormItem, 'inner', {
            props: {
                detail: {},
                formInputs: {},
                form: { type: 'date', text: 'id' },
                value: 'test',
                validateErrors: [],
                increment: 0,
            },
        }), setupLocalVue({
            store: new Vuex.Store({}),
        }));
        expect(wrapper.element).toMatchSnapshot('date form');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(1);
    });

    it('should not render search form', () => {
        const wrapper = mount(app(FormItem, 'inner', {
            props: {
                detail: {},
                formInputs: {},
                form: { type: 'search', search: '::', icon: 'icon', text: 'id' },
                value: 'test',
                validateErrors: [],
                increment: 0,
            },
        }), setupLocalVue({
            store: new Vuex.Store({}),
        }));
        expect(wrapper.element).toMatchSnapshot('search form');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(1);
    });

    it('should not render text form 1', () => {
        const wrapper = mount(app(FormItem, 'inner', {
            props: {
                detail: {},
                formInputs: {},
                form: { text: 'id' },
                value: 'test',
                validateErrors: [],
                increment: 0,
            },
        }), setupLocalVue({
            store: new Vuex.Store({}),
        }));
        expect(wrapper.element).toMatchSnapshot('text form 1');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(1);
    });

    it('should not render text form 2', () => {
        const wrapper = mount(app(FormItem, 'inner', {
            props: {
                detail: {},
                formInputs: {},
                form: { hint: 'column.id', text: 'id' },
                value: 123,
                validateErrors: [],
                increment: 0,
            },
        }), setupLocalVue({
            store: new Vuex.Store({}),
        }));
        expect(wrapper.element).toMatchSnapshot('text form 2');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(1);
    });

    it('should not render time form', () => {
        const wrapper = mount(app(FormItem, 'inner', {
            props: {
                detail: {},
                formInputs: {
                    'reservations.checkout': '10:00:00',
                },
                form: { type: 'time', text: 'id' },
                value: 'test',
                validateErrors: [],
                increment: 0,
            },
        }), setupLocalVue({
            store: new Vuex.Store({}),
        }));
        expect(wrapper.element).toMatchSnapshot('time form');

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(1);
    });
});
