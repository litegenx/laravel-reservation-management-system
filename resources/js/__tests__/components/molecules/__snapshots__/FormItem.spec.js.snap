// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem should not render date form: date form 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div>
          <div
            class="v-dialog__container"
            hint=""
            role="dialog"
          >
            <div
              class="v-input v-input--is-label-active v-input--is-dirty v-input--is-disabled v-input--is-readonly theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate v-icon--disabled material-icons theme--light"
                  >
                    event
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active v-label--is-disabled theme--light"
                      for="input-13"
                      style="left: 0px; position: absolute;"
                    >
                      ID
                    </label>
                    <input
                      disabled="disabled"
                      id="input-13"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-dialog__content"
              role="document"
              style="z-index: 0;"
            >
              <div
                class="v-dialog"
                style="max-width: 850px; display: none;"
              />
            </div>
          </div>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`FormItem should not render search form: search form 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div>
          <div
            hint=""
          >
            <div
              class="v-input v-input--is-readonly theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate material-icons theme--light"
                  >
                    icon
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-29"
                      style="left: 0px; position: absolute;"
                    >
                      ID
                    </label>
                    <input
                      id="input-29"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-dialog__container"
              role="dialog"
            >
              <div
                class="v-dialog__content"
                role="document"
                style="z-index: 0;"
              >
                <div
                  class="v-dialog"
                  style="width: 400px; display: none;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`FormItem should not render text form 1: text form 1 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div>
          <div
            class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            detail="[object Object]"
            form="[object Object]"
            form-inputs="[object Object]"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label v-label--active theme--light"
                    for="input-46"
                    style="left: 0px; position: absolute;"
                  >
                    ID
                  </label>
                  <input
                    id="input-46"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`FormItem should not render text form 2: text form 2 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div>
          <div
            class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field"
            detail="[object Object]"
            form="[object Object]"
            form-inputs="[object Object]"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label v-label--active theme--light"
                    for="input-59"
                    style="left: 0px; position: absolute;"
                  >
                    ID
                  </label>
                  <input
                    id="input-59"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  >
                    <div
                      class="v-messages__message"
                    >
                      ID
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`FormItem should not render time form: time form 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div>
          <div
            class="v-dialog__container"
            detail="[object Object]"
            form="[object Object]"
            hint=""
            role="dialog"
          >
            <div
              class="v-input v-input--is-label-active v-input--is-dirty v-input--is-disabled v-input--is-readonly theme--light v-text-field"
            >
              <div
                class="v-input__prepend-outer"
              >
                <div
                  class="v-input__icon v-input__icon--prepend"
                >
                  <i
                    aria-hidden="true"
                    class="v-icon notranslate v-icon--disabled material-icons theme--light"
                  >
                    timer
                  </i>
                </div>
              </div>
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active v-label--is-disabled theme--light"
                      for="input-73"
                      style="left: 0px; position: absolute;"
                    >
                      ID
                    </label>
                    <input
                      disabled="disabled"
                      id="input-73"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="v-dialog__content"
              role="document"
              style="z-index: 0;"
            >
              <div
                class="v-dialog time-picker-dialog"
                style="max-width: 850px; display: none;"
              />
            </div>
          </div>
        </div>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
